### PR TITLE
Add starting point for multi-language support + Japanese locale

### LINKED ISSUE #3888: Add Japanese language to TTS
#### Summary
With the amazing help of Hidetoshi Yoshida and Koji Aoya of the Japanese TTN community, we have obtained a Japanese translation of TTS. We need to do some preparation to add this to the code.\

#### Why do we need this?
To improve the experience for Japanese speaking TTS users.

#### What is already there? What do you see now?
A set of translated messages.

#### What is missing? What do you want to see?
- Some general revisit of our i18n workflow (see https://github.com/TheThingsNetwork/lorawan-stack/issues/2435#issuecomment-790708961)
- Japanese locale

#### How do you propose to implement this?
- Adjust the `mg js:clean` command to only delete `en.json` and `xx.json`
- Add `ja` to the list of supported languages in our babel config

#### How do you propose to test this?
We can add end-to-end tests for this.

#### Can you do this yourself and submit a Pull Request?
Yes.

### PR DESCRIPTION
**Pull request by kschiffer**
#### Summary
This PR will add first steps towards a more systematic i18n setup for the frontend. It also adds the Japanese locale and hence Japanese language support to the web UIs.

Japanese locale was provided by Japanese TTN Community members Hidetoshi Yoshida and  Koji  Aoya. ありがとうございました!

image

image

References #2435
Closes #3888 

#### Changes
- Refactor `UserLocale`/ `with-locale` component
  - Turn into a functional component with hooks
  - Refactor message loading mechanism
  - Add a Language context for child components to control language
  - Add functionality to persist preferred language in `LocalStorage`
- Revisit locale loading logic
- Make build process more deterministic wrt language and locale support
  - Source supported languages directly from the `locales` folder
  - Remove bundling English messages directly into main bundle
  - Remove `xx` locale logic
- Resolve an issue in the build process that caused many unnecessary locale-related files to be emitted
- Resolve wrong use of webpack magic comments
- Refactor `<Footer />` component
  - Add language chooser
  - Improve styling of footer elements
  - Improve responsive styles
  - Change left side claim into copyright colophon
- Remove `xx` locale and `xx` locale generation
- Add Japanese locale

#### Testing

Manual testing.

##### Regressions

Since this changes the `<UserLocale />` component quite a bit it's possible that this breaks i18n message loading in some scenarios.

#### Notes for Reviewers

This issue is not closing #2435 because there are still some tasks missing. I will lay out the next steps in the issue once this is merged.

#### Checklist

- [x] Scope: The referenced issue is addressed, there are no unrelated changes.
- [x] Compatibility: The changes are backwards compatible with existing API, storage, configuration and CLI, according to the compatibility commitments in `README.md` for the chosen target branch.
- [x] Documentation: Relevant documentation is added or updated.
- [x] Changelog: Significant features, behavior changes, deprecations and fixes are added to `CHANGELOG.md`.
- [x] Commits: Commit messages follow guidelines in `CONTRIBUTING.md`, there are no fixup commits left.

### CODE REVIEW

**Review by adriansmares — APPROVED** [review]
Code-owned file LGTM.

File: pkg/webui/components/icon/icon.styl [hunk 1]
```diff
@@ -28,3 +28,9 @@
   &.nudgeDown
     position: relative
     top: 1px
+
+.text-padded
+  margin-right: $cs.xxs
+  
+  &-left
+    margin-left: $cs.xxs
```
Comment (bafonins) [inline pkg/webui/components/icon/icon.styl]: new line missing

File: pkg/webui/components/icon/index.js [hunk 2]
```diff
@@ -98,6 +100,8 @@ Icon.propTypes = {
   nudgeUp: PropTypes.bool,
   /** Renders a smaller icon. */
   small: PropTypes.bool,
+  /** Whether icon should be padded for a text displayed next to it. */
+  textPadded: PropTypes.oneOfType([PropTypes.string, PropTypes.bool]),
```
Comment (bafonins) [inline pkg/webui/components/icon/index.js]: Can we have separate props for this?

File: pkg/webui/lib/components/with-locale.js [hunk 3]
```diff
@@ -12,193 +12,179 @@
...
-    }
-  }
-
-  @bind
-  async load(language) {
-    let locale = navigator.language || navigator.browserLanguage || defaultLocale
+  }, [])
+
+  const load = useCallback(
+    async (locale, storeAsPreference = false) => {
```
Comment (bafonins) [inline pkg/webui/lib/components/with-locale.js]:
```suggestion
    async (locale) => {
```

`storeAsPreference` is not used anywhere

File: pkg/webui/lib/components/with-locale.js [hunk 4]
```diff
@@ -12,193 +12,179 @@
...
+        setLocale(locale)
+        if (language === defaultLanguage) {
+          // Merge frontend and backend messages.
+          setMessages({ ...res[0], ...res[1] })
+        } else {
+          setMessages(res[0])
+        }
+        localStorage.setItem('lang', language)
+      } catch (err) {
+        handleIntlError(err)
```
Comment (bafonins) [inline pkg/webui/lib/components/with-locale.js:180]: If something goes wrong in `try` section, the Console hangs with the spinner. I think we should display an error or proceed with default locale.

**Review by bafonins — APPROVED** [review]
Works well! Please see my comments

**Comment by johanstokking** [comment]
Nice!

**Review by htdvisser — APPROVED** [review]

**Pull request merged**